core/templates/dev/head/domain/question/QuestionPlayerBackendApiService.js

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @fileoverview Service to receive questions for practice given a set of
+ * skill_ids.
+ */
+oppia.constant(
+  'QUESTION_PLAYER_URL_TEMPLATE',
+  '/question_player_handler?skill_ids=<skill_ids>&question_count' +
+  '=<question_count>&start_cursor=<start_cursor>');
+
+oppia.factory('QuestionPlayerBackendApiService', [
+  '$http', '$q', 'UrlInterpolationService', 'QUESTION_PLAYER_URL_TEMPLATE',
+  function($http, $q, UrlInterpolationService, QUESTION_PLAYER_URL_TEMPLATE) {
+    var _startCursor = '';
+    var _fetchQuestions = function(
+        skillIds, questionCount, successCallback, errorCallback) {
+      if (!validateRequestParameters(skillIds, questionCount, errorCallback)) {
+        return;
+      }
+      var questionDataUrl = UrlInterpolationService.interpolateUrl(
+        QUESTION_PLAYER_URL_TEMPLATE, {
+          skill_ids: skillIds.join(','),
+          question_count: questionCount.toString(),
+          start_cursor: _startCursor
+        });
+
+      $http.get(questionDataUrl).then(function(response) {
+        var questionDicts = angular.copy(response.data.question_dicts);
+        _startCursor = response.data.next_start_cursor;
```

Comment:
Here, in a browser session, `_startCursor` is never reset, right? 
I am not sure of this, but for example, let's say first the user requests for 2 questions linked to skill_1, and then `_startCursor` would be set to the next cursor value for that set, then if the user requests 2 questions linked to skill_2, won't the same next cursor value be used for this call as well? Whereas here, we ideally want to start over with `_startCursor = ''` again. I'd suggest having a `resetHistory` or some similar flag be passed here as well through `fetchQuestions` (see how `fetchQuestionSummaries()` are done in TopicEditorStateService).
What do you say?

Reply:
I'm not very sure about this. If we want to reset it I guess frontend would have to pass this value to the class. @vinitamurthi what do you think regarding this?

Reply:
What we can do is, when we call the question player interface from practice session, we pass in a startCursor that is blank. Does that work?